android/src/main/java/com/rnfingerprint/FingerprintAuthModule.java

```diff
@@ -0,0 +1,200 @@
+import javax.crypto.NoSuchPaddingException;
+import javax.crypto.SecretKey;
+import java.security.InvalidKeyException;
+import java.security.KeyStoreException;
+import java.security.UnrecoverableKeyException;
+
+
+public class FingerprintAuthModule extends ReactContextBaseJavaModule {
+
+  public static boolean inProgress = false;
+
+  public FingerprintAuthModule(ReactApplicationContext reactContext) {
+    super(reactContext);
+  }
+
+  @Override
+  public String getName() {
+    return "FingerprintAuth";
+  }
+
+  @ReactMethod
+  public void isSupported(Callback reactErrorCallback, Callback reactSuccessCallback) {
+    keyguardManager =
+            (KeyguardManager) getCurrentActivity().getSystemService(Context.KEYGUARD_SERVICE);
+    fingerprintManager =
+            (FingerprintManager) getCurrentActivity().getSystemService(Context.FINGERPRINT_SERVICE);
+    if(!isFingerprintAuthAvailable()) {
+      reactErrorCallback.invoke("Not supported.");
+    } else {
+      reactSuccessCallback.invoke("Is supported.");
+    }
+    return ;
+  }
+
+  @ReactMethod
+  public void authenticate(String reason, ReadableMap authConfig, Callback reactErrorCallback, Callback reactSuccessCallback) {
+    if (!inProgress) {
+      inProgress = true;
+      keyguardManager =
+      (KeyguardManager) getCurrentActivity().getSystemService(Context.KEYGUARD_SERVICE);
+      fingerprintManager =
+      (FingerprintManager) getCurrentActivity().getSystemService(Context.FINGERPRINT_SERVICE);
+
+      Activity activity = getCurrentActivity();
+
+      if (isFingerprintAuthAvailable()) {
+        generateKey();
+        if (cipherInit()) {
+          cryptoObject = new FingerprintManager.CryptoObject(cipher);
+          fingerprintDialog = new FingerprintDialog();
+          fingerprintDialog.setCryptoObject(cryptoObject);
+
+          DialogResultHandler drh = new DialogResultHandler(reactErrorCallback, reactSuccessCallback);
+
+          fingerprintDialog.setReasonForAuthentication(reason);
+          fingerprintDialog.setAuthConfig(authConfig);
+          fingerprintDialog.setDialogCallback(drh);
+
+          fingerprintDialog.show(activity.getFragmentManager(),"fingerprint_dialog");
+        }
+      }
+
+      return;
+
+    }
+  }
+
+  /*** TOUH ID ACTIVITY REALTED STUFF ***/
+  private FingerprintDialog fingerprintDialog;
+
+  private FingerprintManager fingerprintManager;
+  private KeyguardManager keyguardManager;
+
+  private KeyStore keyStore;
+  private KeyGenerator keyGenerator;
+
+  private Cipher cipher;
+  private static final String KEY_NAME = "example_key";
+
+  private FingerprintManager.CryptoObject cryptoObject;
+
+  private Context appContext;
+
+  public boolean isFingerprintAuthAvailable() {
+
+      if (android.os.Build.VERSION.SDK_INT < 23) {
+          return false;
+      }
+
+      if (!keyguardManager.isKeyguardSecure()) {
+          return false;
+      }
+
+      if (ActivityCompat.checkSelfPermission(getReactApplicationContext(),
```

Comment:
You do not need this block, see [USE_FINGERPRINT](https://developer.android.com/reference/android/Manifest.permission.html#USE_FINGERPRINT). This is a ` normal` permission so user cannot disable it.